var/spack/repos/builtin/packages/intel-oneapi-compilers/package.py

```diff
@@ -70,3 +70,8 @@ def setup_run_environment(self, env):
                          'compiler', 'latest', 'linux', 'lib'))
         env.prepend_path('LD_LIBRARY_PATH', join_path(self.prefix,
                          'compiler', 'latest', 'linux', 'lib'))
+        # Set these so that MPI wrappers will pick up these compilers
+        # when this module is loaded.
+        env.set('I_MPI_CC', 'icx')
+        env.set('I_MPI_CXX', 'icpx')
+        env.set('I_MPI_FC', 'ifx')
```

Comment:
This is an interesting choice although it doesn't match typical Spack behavior for setting up the MPI compiler wrappers: usually we leave that up to the MPI instance since it generally feels out-of-place in the compiler module. Furthermore, to get this correct for all possible MPI implementations, we'd have to set all the variables here (e.g. `OMPI_CC`, `MPICH_CC`...).

Do you know if there is a fallback to using `CC` (in which case that part of https://github.com/spack/spack/pull/20717  could be useful)?

Reply:
Afaik Intel mpi is the only mpi that recognizes this set of vars, and it doesn't pick up CC. I'd like to set here because the module will be more bulletproof. If you think it doesn't follow slack design patterns, we can just write the modules by hand. Adding the rpath for libfabric is needed, though.

Reply:
I don't think anything can go wrong with this except in strange cases (e.g. if using an `intel-oneapi-mpi` package not built with `icx` and loading the `intel-oneapi-compilers` module, which could reasonably be considered out-of scope) and I agree that it is helpful in some cases, so I'm inclined to approve it, but I want to think on it a bit more.